nix/tullia.nix

```diff
@@ -0,0 +1,72 @@
+self: system:
+
+let
+  ciInputName = "GitHub event";
+in
+rec {
+  tasks =
+    let
+      inherit (self.inputs.tullia) flakeOutputTasks taskSequence;
+
+      common =
+        { config
+        , ...
+        }: {
+          preset = {
+            # needed on top-level task to set runtime options
+            nix.enable = true;
+
+            github-ci = {
+              # Tullia tasks can run locally or on Cicero.
+              # When no facts are present we know that we are running locally and vice versa.
+              # When running locally, the current directory is already bind-mounted
+              # into the container, so we don't need to fetch the source from GitHub
+              # and we don't want to report a GitHub status.
+              enable = config.actionRun.facts != { };
+              repo = "input-output-hk/plutus";
+              sha = config.preset.github-ci.lib.getRevision ciInputName null;
+            };
+          };
+        };
+
+      ciTasks = __mapAttrs
+        (_: flakeOutputTask: { ... }: {
+          imports = [ common flakeOutputTask ];
+
+          memory = 1024 * 8;
+          nomad.resources.cpu = 10000;
+        })
+        (flakeOutputTasks [ "ciJobs" system ] self);
+
+      ciTasksSeq = taskSequence "ci/" ciTasks (__attrNames ciTasks);
```

Comment:
Also, does this mean that if a task fails, none of the ones after it will be run, even if they would succeed? That's _significantly_ worse for diagnosing issues than the current setup. 

Could we at least make it so that the different systems are run independently? They shouldn't share any deps so there won't be any risk of redoing work, and that's one of the most common ways for a subset of the builds to fail (system X is broken but not system Y).

Reply:
I believe the other tasks will keep running, the one that failed will just report a github status failure. @dermetfan can you confirm this?

And I think its already setup so different systems run independently. The `tullia` output has seperate exports for each system.

Reply:
Unfortunately no, @michaelpj's concern is correct. We could start all tasks at the same time, in which case they would all finish independently of each other, but I purposely did not do that to keep the needed memory down. Maybe that's not necessary here, we could try.

Reply:
Well, there's also the worry about them redoing work, since they'll all have shared Nix dependencies. I don't mind too much about them running sequentially, but it would be nice if we got failure reports for everything. Although actually maybe that's a bad idea: if there's a failing nix derivation that they all depend on, then trying to keep going would just result in building the same derivation over and over and failing each time, which could be bad/expensive. Hydra handles this more gracefully...